src/Microsoft.SqlTools.ServiceLayer/LanguageServices/BindingQueue.cs

```diff
@@ -370,49 +370,44 @@ private void ProcessQueue()
                                             }
 
                                             RemoveBindingContext(queueItem.Key);
-                                        }    
+                                        }
                                     }
-                                });
+                                }, new CancellationTokenSource(bindTimeout).Token);
 
-                                Task.Run(() => 
+                                try
                                 {
-                                    try
-                                    {
-                                        // check if the binding tasks completed within the binding timeout                           
-                                        if (bindTask.Wait(bindTimeout))
-                                        {
-                                            queueItem.Result = result;
-                                        }
-                                        else
+                                    await bindTask.ContinueWithOnFaulted(t => Logger.Error("Binding queue threw exception " + t.Exception.ToString()))
+                                        .ContinueWith(t =>
                                         {
-                                            cancelToken.Cancel();
+                                            if (t.IsCanceled)
```

Comment:
When would the task itself be cancelled outside of the cancelToken already being cancelled?

Reply:
Do you want me to update `cancelToken` to cancel on `bindTimeoutInMs` instead?
Currently cancelToken doesn't have a timeout attached.